app/reflexes/invite_manager_reflex.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class InviteManagerReflex < ApplicationReflex
+  include ManagerInvitations
+
+  def invite
+    email = params[:email]
+    enterprise = Enterprise.find(params[:enterprise_id])
+
+    authorize! :edit, enterprise
+
+    existing_user = Spree::User.find_by(email: email)
+
+    locals = { error: nil, success: nil, email: email, enterprise: enterprise }
+
+    if existing_user
+      locals[:error] = I18n.t('admin.enterprises.invite_manager.user_already_exists')
+
+      return_morph(locals)
+      return
+    end
+
+    new_user = create_new_manager(email, enterprise)
+
+    if new_user.valid?
+      locals[:success] = true
+    else
+      locals[:error] = new_user.errors.full_messages.to_sentence ||
+                       I18n.t('admin.enterprises.invite_manager.error')
```

Comment:
Instead of running the validations again, it's enough to check for errors. That's actually what you need for the `else` part. And then you don't need a backup error message.
```suggestion
    if new_user.errors.empty?
      locals[:success] = true
    else
      locals[:error] = new_user.errors.full_messages.to_sentence
```

Reply:
Thanks, that's better!